pylint/message/message.py

```diff
@@ -41,33 +43,30 @@ def __new__(
         symbol: str,
         location: MessageLocationTuple,
         msg: str,
-        confidence: Optional[Confidence],
-    ) -> "Message":
+        confidence: Confidence | None,
+    ) -> Message:
         ...
 
     @overload
     def __new__(
         cls,
         msg_id: str,
         symbol: str,
-        location: Tuple[str, str, str, str, int, int],
+        location: tuple[str, str, str, str, int, int],
         msg: str,
-        confidence: Optional[Confidence],
-    ) -> "Message":
+        confidence: Confidence | None,
+    ) -> Message:
         # Remove for pylint 3.0
         ...
 
     def __new__(
         cls,
         msg_id: str,
         symbol: str,
-        location: Union[
-            Tuple[str, str, str, str, int, int],
-            MessageLocationTuple,
-        ],
+        location: (tuple[str, str, str, str, int, int] | MessageLocationTuple),
```

Comment:
```suggestion
        location: MessageLocationTuple,
```
Could we do this ? It seems those should be the same thing.

Reply:
No, see the deprecation warning a couple of lines below. `location` needs to be initialised as `MessageLocationTuple` and not be a regular `tuple`.